jacodb-analysis/src/main/kotlin/org/jacodb/analysis/config/Condition.kt

```diff
@@ -0,0 +1,197 @@
+    }
+
+    override fun visit(condition: Not): Boolean {
+        return !condition.arg.accept(this)
+    }
+
+    override fun visit(condition: ConstantTrue): Boolean {
+        return true
+    }
+
+    override fun visit(condition: IsConstant): Boolean {
+        val value = positionResolver.resolve(condition.position)
+        return value is JcConstant
+    }
+
+    override fun visit(condition: IsType): Boolean {
+        error("Unexpected condition: $condition")
+    }
+
+    override fun visit(condition: AnnotationType): Boolean {
+        error("Unexpected condition: $condition")
+    }
+
+    override fun visit(condition: ConstantEq): Boolean {
+        val value = positionResolver.resolve(condition.position)
+        return when (val constant = condition.value) {
+            is ConstantBooleanValue -> {
+                value is JcBool && value.value == constant.value
+            }
+
+            is ConstantIntValue -> {
+                value is JcInt && value.value == constant.value
+            }
+
+            is ConstantStringValue -> {
+                // TODO: if 'value' is not string, convert it to string and compare with 'constant.value'
+                value is JcStringConstant && value.value == constant.value
+            }
+
+            else -> error("Unexpected constant: $constant")
+        }
+    }
+
+    override fun visit(condition: ConstantLt): Boolean {
+        val value = positionResolver.resolve(condition.position)
+        return when (val constant = condition.value) {
+            is ConstantIntValue -> {
+                value is JcInt && value.value < constant.value
+            }
+
+            else -> error("Unexpected constant: $constant")
+        }
+    }
+
+    override fun visit(condition: ConstantGt): Boolean {
+        val value = positionResolver.resolve(condition.position)
+        return when (val constant = condition.value) {
+            is ConstantIntValue -> {
+                value is JcInt && value.value > constant.value
+            }
+
+            else -> error("Unexpected constant: $constant")
+        }
+    }
+
+    override fun visit(condition: ConstantMatches): Boolean {
+        val value = positionResolver.resolve(condition.position)
+        val re = condition.pattern.toRegex()
+        return re.matches(value.toString())
+    }
+
+    override fun visit(condition: SourceFunctionMatches): Boolean {
+        TODO("Not implemented yet")
+    }
+
+    override fun visit(condition: ContainsMark): Boolean {
+        error("This visitor does not support condition $condition. Use FactAwareConditionEvaluator instead")
+    }
+
+    override fun visit(condition: TypeMatches): Boolean {
+        val value = positionResolver.resolve(condition.position)
+        return value.type.isAssignable(condition.type)
+    }
+}
+
+class FactAwareConditionEvaluator(
+    private val fact: Tainted,
+    private val basicConditionEvaluator: BasicConditionEvaluator,
+) : ConditionVisitor<Boolean> {
```

Comment:
```suggestion
) : ConditionVisitor<Boolean> by basicConditionEvaluator {
```
and remove useless overrides

Reply:
Agree, done